File: src/dungeon/room.rs
/// Room definition: position, dimensions, exits
#[derive(Debug, Clone, PartialEq, Eq)]
pub struct Room {
    pub id: usize,
    pub x: u32,
    pub y: u32,
    pub width: u32,
    pub height: u32,
    pub room_type: RoomType,
    pub exits: Vec<Exit>,
    pub parent_id: Option<usize>,  // Which room this connects to
    pub parent_wall: Option<Wall>,  // Which wall of parent this connects through
}

#[derive(Debug, Clone, Copy, PartialEq, Eq)]
pub enum RoomType {
    Entrance,
    Normal,
    Small,
    Large,
    Corridor,
}

#[derive(Debug, Clone, PartialEq, Eq)]
pub struct Exit {
    pub wall: Wall,
    pub position: u32,
    pub connected_room_id: Option<usize>,  // Track which room this exit leads to
}

#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
pub enum Wall {
    North,
    East,
    South,
    West,
}

impl Room {
    /// Calculate room area - used by tests and generation logic
    #[allow(dead_code)]
    pub fn area(&self) -> u32 {
        self.width * self.height
    }

    /// Check if room is a corridor (1-width dimension)
    #[allow(dead_code)]
    pub fn is_corridor(&self) -> bool {
        self.width == 1 || self.height == 1
    }
}

File: src/Models/Room.cs
using DungeonSaver.Utils;

namespace DungeonSaver.Models;

public enum RoomType
{
    Entrance,
    Normal,
    Small,
    Large,
    Corridor
}

/// <summary>
/// Represents a room in the dungeon
/// </summary>
public class Room
{
    public int Id { get; set; }
    public Rectangle Bounds { get; set; }
    public RoomType Type { get; set; }
    public List<Exit> Exits { get; set; }
    public bool IsExplored { get; set; }
    public bool IsVisible { get; set; }
    public HashSet<Point> RevealedTiles { get; } = new();

    public Room(int id, Rectangle bounds, RoomType type)
    {
        Id = id;
        Bounds = bounds;
        Type = type;
        Exits = new List<Exit>();
        IsExplored = false;
        IsVisible = false;
    }

    public int Area => Bounds.Area;
    
    public bool Contains(Point point) => Bounds.Contains(point);
    
    public bool Intersects(Room other) => Bounds.Intersects(other.Bounds);
}
